### ISSUE TITLE
Single card(RTX 3090) training results

### ISSUE THREAD

**Issue by sjhfdl** [hustvl/MapTR#65]
Thanks for the great work and sharing the code!

I trained through a single RTX3090 graphics card, the configuration file is maptr_tiny_r50_24e.py, the results after training are shown in the figure, the results are not ideal, and there is a big gap between the results in the paper, I would like to ask if you have tried the single card training, or what should be paid attention to in the training
![Figure_1](https://github.com/hustvl/MapTR/assets/135950499/a9089fd6-6e86-4845-a61a-f455f6669eef)

**Comment by qiuwenzhao** [comment]
We trained it on the two A100 GPUS, and the  Map result is about 0.35 in the epoch 24

**Comment by sjhfdl** [comment]
> We trained it on the two A100 GPUS, and the Map result is about 0.35 in the epoch 24

Hello, I solved this problem, the reason is that the paper used 8 Gpus for training, and I trained on a single card, so I reduced the initial learning rate lr and weight_decay by 8 times, changed to lr=0.75e-4, weight_decay=0.00125, and then decreased the initial learning rate LR and weight_decay by 8 times. Also, enlarge the warmup_iters in lr_config by a factor of eight, to 4000

**Comment by adasfag** [comment]
> 

Hello .I train the model again as your advice,but the Map is about 45.7 in the epoch 24. Could you provicd your single GPU result?Thank you very much

**Comment by sjhfdl** [comment]
> > 
> 
> Hello .I train the model again as your advice,but the Map is about 45.7 in the epoch 24. Could you provicd your single GPU result?Thank you very much

First of all, I would like to apologize to you. Due to the computing power of my graphics card, when I adjusted the learning rate, I only trained the author's code for two epochs, and I felt that the accuracy of the second epoch had reached 0.15, so I did not continue the training. Then I went to verify my method, and the accuracy of the training was similar to the results given by the author. My idea is that the results of the multi-card run will be slightly lower than those of the single card, and then I assume that the method of the author can also run on my own computer and produce similar results as in the paper.

**Comment by adasfag** [comment]
> > > 
> > 
> > 
> > Hello .I train the model again as your advice,but the Map is about 45.7 in the epoch 24. Could you provicd your single GPU result?Thank you very much
> 
> First of all, I would like to apologize to you. Due to the computing power of my graphics card, when I adjusted the learning rate, I only trained the author's code for two epochs, and I felt that the accuracy of the second epoch had reached 0.15, so I did not continue the training. Then I went to verify my method, and the accuracy of the training was similar to the results given by the author. My idea is that the results of the multi-card run will be slightly lower than those of the single card, and then I assume that the method of the author can also run on my own computer and produce similar results as in the paper.

It seems that the single result is lower than those of the multi-card, maybe it needs a more suitable lr and It confused me.

**Comment by sjhfdl** [comment]
> It seems that the single result is lower than those of the multi-card, maybe it needs a more suitable lr and It confused me.

Yes, you need a good learning rate configuration, you can try it a few times, maybe because our graphics card models are different

**Comment by sjhfdl** [comment]
> > > 
> > 
> > 
> > Hello .I train the model again as your advice,but the Map is about 45.7 in the epoch 24. Could you provicd your single GPU result?Thank you very much
> 
> First of all, I would like to apologize to you. Due to the computing power of my graphics card, when I adjusted the learning rate, I only trained the author's code for two epochs, and I felt that the accuracy of the second epoch had reached 0.15, so I did not continue the training. Then I went to verify my method, and the accuracy of the training was similar to the results given by the author. My idea is that the results of the multi-card run will be slightly lower than those of the single card, and then I assume that the method of the author can also run on my own computer and produce similar results as in the paper.

Could you show me the test results of the training？

**Comment by sjhfdl** [comment]
> > 
> 
> Hello .I train the model again as your advice,but the Map is about 45.7 in the epoch 24. Could you provicd your single GPU result?Thank you very much

Could you show me the test results of the training？

**Comment by adasfag** [comment]
> 

<img width="383" alt="0 214-0 5" src="https://github.com/hustvl/MapTR/assets/73413158/5e689263-1946-4134-a195-e944e5b789ea">
<img width="392" alt="0 498-1 0" src="https://github.com/hustvl/MapTR/assets/73413158/95a5bb53-3473-4632-863f-6f847ec3b119">
<img width="400" alt="0 659-1 0" src="https://github.com/hustvl/MapTR/assets/73413158/0a641fe9-8f90-4391-8435-bcffaf6cbd75">

**Comment by Sweetylrx** [comment]
I have met this problem as well

**Comment by PlutoXN** [comment]
> 感谢您的出色工作和分享代码！
> 
> 我通过单块RTX3090显卡进行训练，配置文件为maptr_tiny_r50_24e.py，训练后的结果如图所示，结果不太理想，和论文中的结果有很大差距，我想请问您是否尝试过单卡训练，或者训练中需要注意什么 ![图1](https://user-images.githubusercontent.com/135950499/244295111-a9089fd6-6e86-4845-a61a-f455f6669eef.png)

Hello, how to set up single card training? Can you share the steps on how to modify it?

**Comment by VanHelen** [comment]
Hello，have you solve this problem？I have trained through a single RTX4090, the configuration file is maptr_tiny_r50_24e.py, but it alwalys stop training at epoch 10, with the following problem 
![image](https://github.com/user-attachments/assets/fa154307-63c9-4cfc-b36e-d6c6070bbb75)
May I ask what are the parameters required for completing 24 epochs of training?

**Comment by VanHelen** [comment]
> Hello guys, I wonder how much time you spend with a single card?

Hello，have you successfully trained the model on a single GPU？May I ask what parameters need to be modified? I am currently experiencing training failure at epoch 10. 
![image](https://github.com/user-attachments/assets/36f98985-9cd9-466e-a65f-d599ce3a68b6)
